tests/macsec/test_reboot.py

```diff
@@ -0,0 +1,71 @@
+'''This script is to test the device reboot behavior for SONiC.
+
+Test 1: Line Card
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on DUT and reboot the line card
+Step 3: Verify macsec connection is re-established after reboot
+
+Test 2: Chassis
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on DUT and reboot the chassis
+Step 3: Verify macsec connection is re-established after reboot
+
+'''
+
+import logging
+import pytest
+# from time import sleep
+from tests.common import reboot
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+
+@pytest.mark.reboot
+def test_lc_reboot(duthost, localhost):
+    # duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    dut_lldp_table = duthost.shell("show lldp table")['stdout'].split("\n")
+    dut_to_neigh_int = dut_lldp_table[3].split()[0]
```

Comment:
Can we make it more reobost, instead of checking only one interface. Can we also do 
1. Check/store in a dict all macsec sessions up and enabled before reboot
2. After reboot, confirm if all the macsec sessions are up

Reply:
@Azarack , Please address this comment.

Reply:
The test now gathers the macsec session list before and after the reboot and compares to ensure they're the same.